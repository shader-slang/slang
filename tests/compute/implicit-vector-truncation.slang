//TEST(compute):COMPARE_COMPUTE_EX:-slang -compute
//TEST(compute):COMPARE_COMPUTE_EX:-slang -compute -dx12
//TEST(compute, vulkan):COMPARE_COMPUTE_EX:-vk -compute

int sum(int3 v)
{
	return v.x + v.y + v.z;
}

int test(int val)
{
	int4 a = int4(val, val*0x10, val*0x100, val*0x1000);
	return sum(a);
}

//TEST_INPUT:ubuffer(data=[0 0 0 0], stride=4):dxbinding(0),glbinding(0),out
RWStructuredBuffer<int> gBuffer;

[numthreads(4, 1, 1)]
void computeMain(uint3 dispatchThreadID : SV_DispatchThreadID)
{
	uint tid = dispatchThreadID.x;
	int val = int(tid);
	val = test(val);
	gBuffer[tid] = val;
}