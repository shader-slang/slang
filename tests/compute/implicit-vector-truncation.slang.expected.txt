0
111
222
333
